pkgs/applications/virtualization/docker/default.nix

```diff
@@ -15,10 +15,10 @@ rec {
       , iptables, e2fsprogs, xz, util-linux, xfsprogs, git
       , procps, rootlesskit, slirp4netns, fuse-overlayfs, nixosTests
       , clientOnly ? !stdenv.isLinux, symlinkJoin
-      , withSystemd ? true, systemd
-      , withBtrfs ? true, btrfs-progs
+      , withSystemd ? stdenv.isLinux, systemd
+      , withBtrfs ? stdenv.isLinux, btrfs-progs
       , withLvm ? true, lvm2
```

Comment:
```suggestion
      , withLvm ? stdenv.isLinux, lvm2
```

Reply:
but that builds on darwin.

Reply:
are you sure that's the case? https://github.com/NixOS/nixpkgs/blob/a115bb9bd56831941be3776c8a94005867f316a7/pkgs/os-specific/linux/lvm2/common.nix#L168
the original reply also implies so https://github.com/NixOS/nixpkgs/pull/202437#discussion_r1033954636

Reply:
It's in the build results for this PR as well:

* https://logs.nix.ci/?key=nixos/nixpkgs.203535&attempt_id=04a33842-6b3a-4250-a7bb-33533f3daf07

```
Cannot nix-instantiate `docker' because:
error: Package ‘lvm2-2.03.16’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/macstadium-m1-44911104/pkgs/os-specific/linux/lvm2/common.nix:167 is not supported on ‘aarch64-darwin’, refusing to evaluate.
```

Reply:
With or without the changes from this PR, I get `error: Package ‘lvm2-2.03.16’ in /nix/store/schx4dxph7ig4xflaqzx3wpqbl75js4y-nixpkgs/nixpkgs/pkgs/os-specific/linux/lvm2/common.nix:167 is not supported on ‘x86_64-darwin’, refusing to evaluate.` trying to build docker on darwin.

With the extra change suggested by figsoda, it builds successfully.

Reply:
How is ofborg green when refusing to evaluate!?!?

Reply:
That's how it skips tests when `passthru.tests` doesn't exist